pkg/destroy/azure/azure.go

```diff
@@ -419,17 +423,63 @@ func deleteAzureStackPublicRecords(ctx context.Context, o *ClusterUninstaller) e
 	return utilerrors.NewAggregate(errs)
 }
 
-func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient, privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+func deleteRecordsFromBaseDomain(ctx context.Context, o *ClusterUninstaller) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	if o.BaseDomainResourceGroupName == "" || o.ZoneName == "" || o.APIURL == "" {
+		o.Logger.Debugf("could not find values in the metadata to get record set")
+		return nil
+	}
+
+	var errs []error
+	tag := fmt.Sprintf("kubernetes.io_cluster.%s", o.InfraID)
+	result, err := o.recordsClient.Get(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME)
```

Comment:
Shall we also take `*.apps` dns records into consideration ?

Reply:
We should thanks! Fixed it.

Reply:
I don't see a check for *.apps. we probably need to wrap this in a loop to check for both api & *.apps

Reply:
Wrong commit sorry. Fixed it